### ISSUE TITLE
Hot Loader Clears Store!

### ISSUE THREAD

**Issue by dcflow** [solkimicreb/react-easy-state#40]
Hello,

I'm using `react-hot-loader` on my project and I have the `store` objects declared outside of components.
Every time I do a small change on any of the main or child components the store clears up to default state.
Is this how it is supposed to be .. this is the same as doing a full reload.

Small example with create-react-app
```
import React from "react";
import { hot } from "react-hot-loader";
import { store, view } from "react-easy-state";

const storage = store({ test: 0 });

class App extends React.Component {
  componentDidMount() {
    storage.test = 1;
  }
  render() {
    console.log("App");
    return <h1>App {storage.test}</h1>;
  }
}

export default hot(module)(view(App));
```
When I change the header for example from `App` to `App2` the store returns to default.

Thank You!

**Comment by solkimicreb** [comment]
Yep, that can happen, if you have your stores and components in the same file. Try to move them to separate files. If you have them nicely separated, editing your components won't clear the state. You won't even need hot-loader, vanilla webpack HMR is enough.

(Just add this to your index.js and you are good to go)

```js
if (module.hot) {
  module.hot.accept('./App', () => {
    ReactDOM.render(<App />, document.getElementById('root'));
  });
}
```

**Comment by dcflow** [comment]
Moving the store to another file works great. 👍 
This is so funny .. look at the issue I opened at [react-hot-loader](https://github.com/gaearon/react-hot-loader/issues/930) this morning. 
I followed the instructions and they say to use `webpack HMR` + `react-hot-loader` `plugin`. 
Now that I read your comment I was like yeah why use them both .. it turns out they both do a update of the page and the `render` method is called twice. 😄 
The price we pay from not writing an extra couple lines of code. :man_facepalming:
